src/Aspire.Hosting.Redis/RedisBuilderExtensions.cs

```diff
@@ -19,18 +19,124 @@ public static class RedisBuilderExtensions
+    /// 
+    /// the example of adding <see cref="RedisResource"/> to application model.
+    /// <code>
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    ///
+    /// var redis = builder.AddRedis("redis",6379);
+    /// 
+    /// var api = builder.AddProject&lt;Projects.Api&gt;("api")
+    ///   .WithReference(redis);
+    ///   
+    /// builder.Build().Run(); 
+    /// </code>
+    ///
+    /// </example>
+    public static IResourceBuilder<RedisResource> AddRedis(this IDistributedApplicationBuilder builder, string name, int? port)
     {
-        var redis = new RedisResource(name);
-        return builder.AddResource(redis)
+        return builder.AddRedis(name, port, null);
+    }
+
+    /// <summary>
+    /// Adds a Redis container to the application model.
+    /// </summary>
+    /// <remarks>
+    /// The default image is "redis" and the tag is "7.2.4".
+    /// If a password parameter is not supplied, a password resource will be added to the model at publish time.
+    /// </remarks>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="port">The host port to bind the underlying container to.</param>
+    /// <param name="password">The parameter used to provide the password for the Redis resource.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    /// <example>
+    /// Use in application host
+    /// <list type="bullet">
+    /// <item>
+    /// the example of adding <see cref="RedisResource"/> to application model without password
+    /// <code>
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    ///
+    /// var redis = builder.AddRedis("redis");
+    /// 
+    /// var api = builder.AddProject&lt;Projects.Api&gt;("api")
+    ///   .WithReference(redis);
+    ///   
+    /// builder.Build().Run(); 
+    /// </code>
+    /// </item>
+    /// <item>
+    /// the example of adding <see cref="RedisResource"/> to application model with password
+    /// <code>
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    /// 
+    /// builder.Configuration["Parameters:pass"] = "StrongPassword";
+    /// var password = builder.AddParameter("pass");
+    /// 
+    /// var redis = builder.AddRedis("redis", password: password);
+    ///
+    /// var api = builder.AddProject&lt;Projects.Api&gt;("api")
+    ///   .WithReference(redis);
+    ///   
+    /// builder.Build().Run(); 
+    /// </code>
+    /// </item>
+    /// </list>
+    /// </example>
+    public static IResourceBuilder<RedisResource> AddRedis(
+        this IDistributedApplicationBuilder builder,
+        string name,
+        int? port = null,
+        IResourceBuilder<ParameterResource>? password = null)
+    {
+        if (password?.Resource is not null)
```

Comment:
@DamianEdwards @mitchdenny @davidfowl - what do you think about not having this logic and just always adding a password if it is null here? Yes, it is technically a behavior breaking change. But it will make Redis consistent with the others that have passwords - mysql, postgres, etc.

Reply:
I agree let's shoot for consistency here.

Reply:
so, we don't need to check the execution context, always adding a password to the resource whether the app is in publish mode or run mode. right?

Reply:
If it is in publish mode and the password isn't provided we'll create a synthetic parameter like we do for Postgres.

Reply:
we already did. look at this

```
        if (builder.ExecutionContext.IsPublishMode)
        {
            var passwordParameter = password?.Resource ?? ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter(builder, $"{name}-password");
            var redis = new RedisResource(name, passwordParameter);

            return builder.AddResource(redis)
                      .WithEndpoint(port: port, targetPort: 6379, name: RedisResource.PrimaryEndpointName)
                      .WithImage(RedisContainerImageTags.Image, RedisContainerImageTags.Tag)
                      .WithImageRegistry(RedisContainerImageTags.Registry);
                      .WithImageRegistry(RedisContainerImageTags.Registry)
                      .WithPassword(passwordParameter);
        }
        else if (builder.ExecutionContext.IsRunMode)
        {
            var redis = new RedisResource(name);
            return builder.AddResource(redis)
                          .WithEndpoint(port: port, targetPort: 6379, name: RedisResource.PrimaryEndpointName)
                          .WithImage(RedisContainerImageTags.Image, RedisContainerImageTags.Tag)
                          .WithImageRegistry(RedisContainerImageTags.Registry);
        }
```

Reply:
I've push an update for what I was talking about. We always add a password when calling `.AddRedis("name")` now. Just like other resources like PostgreSQL.